src/earn/EarnPoolInfoScreen.tsx

```diff
@@ -682,6 +743,107 @@ function InfoBottomSheet({
+    </Touchable>
+  )
+}
+
+function BeforeDepositBottomSheet({
+  forwardedRef,
+  token,
+  hasTokensOnSameNetwork,
+  hasTokensOnOtherNetworks,
+  canAdd,
+  exchanges,
+}: {
+  forwardedRef: RefObject<BottomSheetModalRefType>
+  token: TokenBalance
+  hasTokensOnSameNetwork: boolean
+  hasTokensOnOtherNetworks: boolean
+  canAdd: boolean
+  exchanges: ExternalExchangeProvider[]
+}) {
+  const { t } = useTranslation()
+
+  const title = hasTokensOnSameNetwork
+    ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.youNeedTitle', {
+        tokenSymbol: token.symbol,
+        tokenNetwork: NETWORK_NAMES[token.networkId],
+      })
+    : t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.beforeYouCanDepositTitle')
+
+  const swapAndDepositAction = useSwapAndDepositAction({ token, forwardedRef })
+  const crossChainSwapAction = useCrossChainSwapAction({
+    token,
+    title: hasTokensOnSameNetwork
+      ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.action.crossChainSwap')
+      : t('addFundsActions.swap'),
+    details: hasTokensOnSameNetwork
+      ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.crossChainSwapActionDescription', {
+          tokenSymbol: token.symbol,
+        })
+      : t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.swapActionDescription', {
+          tokenSymbol: token.symbol,
+          tokenNetwork: NETWORK_NAMES[token.networkId],
+        }),
+    forwardedRef,
+  })
+  const addAction = useAddAction({ token, forwardedRef })
+  const transferAction = useTransferAction({ token, exchanges, forwardedRef })
+
+  const components: React.JSX.Element[] = []
+  if (hasTokensOnSameNetwork) {
+    components.push(<ActionCard action={swapAndDepositAction} />)
+    components.push(
+      <Text style={styles.actionDetails}>
+        {t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.crossChainAlternativeDescription', {
+          tokenNetwork: NETWORK_NAMES[token.networkId],
+        })}
+      </Text>
+    )
+    if (hasTokensOnOtherNetworks) components.push(<ActionCard action={crossChainSwapAction} />)
+    if (canAdd) components.push(<ActionCard action={addAction} />)
+  } else if (hasTokensOnOtherNetworks) {
+    components.push(<ActionCard action={crossChainSwapAction} />)
+    if (canAdd) components.push(<ActionCard action={addAction} />)
+    components.push(<ActionCard action={transferAction} />)
+  } else {
+    if (canAdd) components.push(<ActionCard action={addAction} />)
+    components.push(<ActionCard action={transferAction} />)
+  }
+
+  return (
+    <BottomSheet
+      forwardedRef={forwardedRef}
+      title={title}
+      description={
+        !hasTokensOnSameNetwork
+          ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.beforeYouCanDepositDescription')
+          : undefined
+      }
+      titleStyle={styles.bottomSheetTitle}
+      testId={'Earn/BeforeDepositBottomSheet'}
+    >
+      <View style={styles.actionsContainer}>{components}</View>
+    </BottomSheet>
+  )
+}
```

Comment:
This conditional render created by pushing to an array seems a bit convoluted. Could we do something like the code below instead?

```suggestion
  function Actions({
    hasTokensOnSameNetwork,
    hasTokensOnOtherNetworks,
    canAdd,
  }: {
    hasTokensOnSameNetwork: boolean
    hasTokensOnOtherNetworks: boolean
    canAdd: boolean
  }) {
    return (
      <View style={styles.actionsContainer}>
        {hasTokensOnSameNetwork && (
          <>
            <ActionCard action={swapAndDepositAction} />
            <Text style={styles.actionDetails}>
              {t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.crossChainAlternativeDescription', {
                tokenNetwork: NETWORK_NAMES[token.networkId],
              })}
            </Text>
          </>
        )}
        {hasTokensOnOtherNetworks && <ActionCard action={crossChainSwapAction} />}
        {canAdd && <ActionCard action={addAction} />}
        { !hasTokensOnSameNetwork && <ActionCard action={transferAction} />}
      </View>
    )
  }

  return (
    <BottomSheet
      forwardedRef={forwardedRef}
      title={title}
      description={
        !hasTokensOnSameNetwork
          ? t('earnFlow.poolInfoScreen.beforeDepositBottomSheet.beforeYouCanDepositDescription')
          : undefined
      }
      titleStyle={styles.bottomSheetTitle}
      testId={'Earn/BeforeDepositBottomSheet'}
    >
      <Actions 
        hasTokensOnSameNetwork={hasTokensOnSameNetwork}
        hasTokensOnOtherNetworks={hasTokensOnOtherNetworks}
        canAdd={canAdd}
      />
    </BottomSheet>
  )
}
```